### PR TITLE
Added mdp-0.93

### PR DESCRIPTION
**Pull request by fredlebel**
A command-line based markdown presentation tool.

### CODE REVIEW

File: pkgs/applications/misc/mdp/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+stdenv.mkDerivation {
+  name = "mdp-0.93";
+
+  src = fetchgit {
+    url = "git://github.com/visit1985/mdp.git";
+    rev = "09d6bd1a8a33fac75a999f0822ec10cb77fbc072";
+    sha256 = "1jx9i5dzz3invs76f72jzv87cm87y2cq747pkg2f3vsni8d21cam";
+  };
+
+  installPhase = "make install PREFIX=$out";
```
Comment (nckx) [inline pkgs/applications/misc/mdp/default.nix]: Not a style expert, but `makeFlags = "PREFIX=$(out)";` seems to be the idiomatic way to do this. Or does that mess up anotherPhase?

File: pkgs/applications/misc/mdp/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  buildInputs = [ ncurses ];
+
+  meta = {
+    homepage = https://github.com/visit1985/mdp;
+    description = "A command-line based markdown presentation tool.";
```
Comment (nckx) [inline pkgs/applications/misc/mdp/default.nix]: Nitpick: [the full stop can go](http://nixos.org/nixpkgs/manual/#idm140737319319584).

Looks good otherwise!

File: pkgs/applications/misc/mdp/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, fetchgit, ncurses }:
+
+stdenv.mkDerivation {
+  name = "mdp-0.93";
+
+  src = fetchgit {
```
Comment (nckx) [inline pkgs/applications/misc/mdp/default.nix]: I was once told to use `fetchFromGitHub` instead. Take that for what it's worth.

**Comment by fredlebel** [comment]
Won't work with $(out) because it's a variable, not a shell command.

**Comment by jagajaga** [comment]
Please squash commits into one, and force push it to update PR.

**Comment by nckx** [comment]
> Won't work with $(out) because it's a variable, not a shell command.

TL;DR: it will, I tested it :-)

`makeFlags` isn't passed through the shell. Perhaps you're thinking of `makeFlagsArray`?

For `make`, `$(foo)` is just a plain variable. `$foo` and `${foo}` will also work, but are in IMO even more confusing because they look just like shell variables. At least with `$(foo)`, it's obvious that `foo` isn't a command and so no shell can be involved.

**Comment by fredlebel** [comment]
I also tried it and I get this error with `$(out)`.

`/nix/store/frmyf8fdq5xhm035mr87wn6kj0c278g7-stdenv/setup: line 829: out: command not found`

**Comment by nckx** [comment]
That's... weird. Like this?

    -  installPhase = "make install PREFIX=$out";
    +  makeFlags = "PREFIX=$(out)";

**Comment by fredlebel** [comment]
Oops, no.  My mistake.  Pushed new change with `makeFlags`.  I definitely prefer this way.  I was still using `installPhase`.

File: pkgs/applications/misc/mdp/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  name = "mdp-0.93";
+
+  src = fetchFromGitHub {
+    owner = "visit1985";
+    repo = "mdp";
+    rev = "09d6bd1a8a33fac75a999f0822ec10cb77fbc072";
+    sha256 = "0ksa0zqzv1yb8nspxp2vww7bp9y99pcma1vx3cixd3qb5y5ljn1n";
+  };
+
+  makeFlags = "PREFIX=$(out)";
```
Comment (jagajaga) [inline pkgs/applications/misc/mdp/default.nix]: Please, remove brackets. Use just $out.

**Comment by nckx** [comment]
Thanks! Travis approves. Pushed.

**Comment by jagajaga** [comment]
@nckx haven't you seen my comment?

File: pkgs/applications/misc/mdp/default.nix (hunk 4 above)
Comment (nckx) [inline pkgs/applications/misc/mdp/default.nix]: Oops. Page didn't refresh here :-/

**Comment by nckx** [comment]
See above. Chromium + github are horrible here atm. Sorry.

FWIW:

   nckx@dm1 ~/Hacking/nixpkgs $ grep -r 'makeFlags = ' pkgs/ | grep '$(out)' | wc -l
   117
   nckx@dm1 ~/Hacking/nixpkgs $ grep -r 'makeFlags = ' pkgs/ | grep '\[' | wc -l
   194
   nckx@dm1 ~/Hacking/nixpkgs $ grep -r 'makeFlags = ' pkgs/ | grep -v '\[' | wc -l
   223

I'll certainly pay attention to that in future, but the tree is almost maliciously inconsistent about such things.

**Comment by jagajaga** [comment]
:crying_cat_face:

**Comment by nckx** [comment]
:bouquet:

**Comment by fredlebel** [comment]
I'll keep that in mind in the future.  And if I update this package.
